components/servo/lib.rs

```diff
@@ -853,6 +854,8 @@ fn create_constellation(
 
     let resource_sender = public_resource_threads.sender();
 
+    let webgpu_thread = webgpu::WebGPUThread::new();
```

Comment:
Couple of high-level design suggestions, which are perhaps best addressed, if at all, in a follow-up:

1. I wonder if you really want to always start the webgpu component on start-up, rather you might want to start-it when it's first needed. 
2. I wonder if you want any script to just be able to start using WebGPU whenever they want by calling methods on `GPU` inside script. Instead I think you want to mediate that capability via the constellation. So when script calls `RequestAdapter`, it should ideally go through the constellation via a `ScriptMsg::RequestAdapter(sender, options)`, and be treated as an untrusted request for a specific capability. It ties into 1 in the sense that the constellation could then start a WebGPU instance, and respond to script with a dedicated means of communication with it(so subsequent messages don't all have to go through the constellation). 
3. The spec doesn't mention it, however I wonder if you don't want some sort of map on the constellation like `HashMap<EventLoop, WebGPU>`, so that each event-loop(read: [agent-cluster](https://html.spec.whatwg.org/#integration-with-the-javascript-agent-cluster-formalism)) has it's own instance. Perhaps its easier to initially only allow one, meaning reject any `ScriptMsg::RequestAdapter` if a script is already using WebGPU. While the spec says "Applications can hold onto multiple adapters at once"(https://gpuweb.github.io/gpuweb/#adapters), that doesn't necessarily means different origins or agent-cluster can use adapters at the same time. 
4. I see a lot of those interfaces in the spec are `Serializeable`, and I wonder if that also shouldn't be restricted to agent-cluster. Shared array buffers are also restricted to agent-cluster, I believe. See https://html.spec.whatwg.org/#can-share-memory-with and https://tc39.es/ecmascript_sharedmem/shmem.html#WebBrowserEmbedding on the other hand I'm not sure if those objects actually "share memory" or not, or how they would be used concurrently by different agents. 

cc @jdm

Reply:
Thanks for pointing out these @gterzian . I've added this list to the WebGPU meta [issue](https://github.com/servo/servo/issues/24706#issuecomment-555491438), so we will not forget about it, and can discuss these things further. As for 1. and 2. we will definitely do that in the near future.